addons/digest/models/digest.py

```diff
@@ -87,11 +87,17 @@ def create(self, vals_list):
 
     def action_subscribe(self):
         if self.env.user.has_group('base.group_user') and self.env.user not in self.user_ids:
-            self.sudo().user_ids |= self.env.user
+            self.sudo()._action_subscribe_users(self.env.user)
+
+    def _action_subscribe_users(self, users):
+        self.user_ids |= users
 
     def action_unsubcribe(self):
         if self.env.user.has_group('base.group_user') and self.env.user in self.user_ids:
-            self.sudo().user_ids -= self.env.user
+            self.sudo()._action_unsubscribe_users(self.env.user)
```

Comment:
Opinion: I would do the opposit, make the private method use `self.sudo().user_ids` and call it after the check

Reply:
Hagrid.

Reply:
:bearded_person: